### PR TITLE
[CT-1326] Get table stringId with explode with limit=3

### PR DESCRIPTION
**Pull request by tomasfejfar**
Jira: CT-1326
KBC: https://github.com/keboola/connection/pull/4913

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

File: tests/Backend/CommonPart1/CreateTableTest.php [hunk 1]
```diff
@@ -250,6 +250,20 @@ public function testTableWithUnsupportedCharactersInNameShouldNotBeCreated(): vo
         }
     }
 
+    public function testBucketWithUnsupportedCharactersInNameShouldNotBeValid(): void
+    {
+        try {
+            $this->_client->createTable(
+                $this->getTestBucketId() . '.rubbish',
```
Comment (jirkasemmler) [inline tests/Backend/CommonPart1/CreateTableTest.php:257]: chvilku mi trvalo, proc je takovej testcase validni, ale jasne - pred upravou by toto proslo, protoze by se udelal explode bez limitu a jako bucketId by se vzal jen $this->getTestBucketId()

File: tests/Backend/CommonPart1/CreateTableTest.php [hunk 2]
```diff
@@ -250,6 +250,20 @@ public function testTableWithUnsupportedCharactersInNameShouldNotBeCreated(): vo
         }
     }
 
+    public function testBucketWithUnsupportedCharactersInNameShouldNotBeValid(): void
+    {
+        try {
+            $this->_client->createTable(
```
Comment (jirkasemmler) [inline tests/Backend/CommonPart1/CreateTableTest.php:256]: nemohl by to byt `createTableAsync` ? at prestaneme pouzivat deprecovane metody

File: tests/Common/MetadataTest.php [hunk 3]
```diff
@@ -117,6 +117,15 @@ public function testBucketMetadata(string $devBranchType, string $userRole): voi
         $bucketDetail = $this->_testClient->getBucket($bucketId);
         $this->assertCount(1, $bucketDetail['metadata']);
         $this->assertMetadataEquals($metadatas[1], $bucketDetail['metadata'][0]);
+
+        // test bucket id with dot
+        try {
+            $metadatas = $metadataApi->postBucketMetadata($bucketId . '.rubbish', $provider, $testMetadata);
+            $this->fail('Bucket id with dot should not be allowed');
+        } catch (ClientException $e) {
+            $this->assertMatchesRegularExpression('/Bucket in.c-API-tests-[a-zA-Z0-9]+\.rubbish not found/', $e->getMessage());
```
Comment (jirkasemmler) [inline tests/Common/MetadataTest.php:126]: prosim jeste assert na stringCode

**Review by jirkasemmler — CHANGES_REQUESTED** [review]
prosim ten async call a assert na stringcody

File: tests/Backend/CommonPart1/CreateTableTest.php (hunk 1 above)
Comment (tomasfejfar) [reply tests/Backend/CommonPart1/CreateTableTest.php:257]: Jo, optimální je ty testy pustit nad masterem, vidět je spadnout a pak na branchí a vidět že projdou :)

File: tests/Common/MetadataTest.php (hunk 3 above)
Comment (tomasfejfar) [reply tests/Common/MetadataTest.php]: Done.

File: tests/Backend/CommonPart1/CreateTableTest.php (hunk 2 above)
Comment (tomasfejfar) [reply tests/Backend/CommonPart1/CreateTableTest.php]: Done.

**Review by jirkasemmler — APPROVED** [review]

**Pull request merged**